docs/technical-reference/architecture.md

```diff
@@ -204,25 +203,61 @@ The argument `form` is a jQuery-wrapped FORM element that will get submitted to
 
 #### File Selection Panel {#file-selection-panel}
-Documentation not currently available
+
+This screen is shown when there are multiple files to choose from when creating a project, for instance after uploading a zip file with multiple files in it. This interface lets the user choose which ones to import.
+Although OpenRefine only supports one table per project so far, it is possible to select multiple files to import. Their contents will be concatenated into a single table.
 
 #### Parsing UI Panel {#parsing-ui-panel}
-Documentation not currently available
+
+The parsing UI panel is shown when importing data into a new project.
+Primarily, it lets the user select in which format the data is, which determines how it is read and transformed into an OpenRefine project. The back-end will try to supply an informed guess for the format using the [format guesser](#formatguesser), but it
+is not uncommon that this initial choice must be overriden by the user.
+
+Beyond this choice of format, the parsing UI panel offers a configuration panel for the chosen importer. This part of the UI can be defined independently for each input format, given that not all options are relevant for all formats. For instance, when
+selecting the "Text file" option, the specific UI of the `LinedBasedImporter` will be shown. This UI is defined in:
+* `main/webapp/modules/core/scripts/index/parser-interfaces/line-based-parser-ui.html`
+* `main/webapp/modules/core/scripts/index/parser-interfaces/line-based-parser-ui.js`
+
+Other importers generally define their own parsing panel as well.
+
+The link between the format's identifier (MIME type), importer (Java class which defines the parsing logic) and parsing options UI (Javascript class that defines the rendering of this options area) is made in the `main/webapp/modules/core/MOD-INF/controller.js` file,
+where those components are registered together in the `ImportingManager`.
 
 ### Server-side Components {#server-side-components}
 
 #### ImportingController {#importingcontroller}
-Documentation not currently available
 
-#### UrlRewriter {#urlrewriter}
-Documentation not currently available
+An importing controller is a component of the back-end which is in charge of the entire importing workflow, from the initial transfer of the raw data to be imported to the created project, with all the configuration steps in between, [as described in the
+earlier section](#importing-controllers). OpenRefine comes with
+a default importing controller which implements this for data coming from:
+* file upload by the user via the web interface
+* upload of textual information using the clipboard import form
+* download of a file by supplying a URL
+
```

Comment:
:::note
The importing controller is responsible for *new project* creation and is not responsible for opening or importing *existing projects*, which is the responsibility of the `ProjectManager` class.
:::

Reply:
This is one area that often has confused a few contributors in the past.  So maybe this note admonition can help clear things up a bit.